INSTALL.md

```diff
@@ -147,7 +157,15 @@ you'll still need to do that yourself.
 
 # Notes for Windows Users
 
-Windows likes to make things difficult.  Common and historical issues:
+Installing `git-filter-repo` via pip into a virtual envirinment is known to
+work well. This makes `git-filter-repo` available as executable script in the
+virtual env.
+
+An alternative is using [pipx](https://pypa.github.io/pipx/) that is made to
+simplify installing and running Python-scripts. It does also work for
+`git-filter-repo` on Windows.
```

Comment:
The first sentence is really bad; it's not merely orthogonal to reality, it's 180 degrees out of phase with it.  We could rewrite this something like:
```
Installing `git-filter-repo` via pip often causes all kinds of problems for users:
 * https://github.com/newren/git-filter-repo/issues/5#issuecomment-618981046
 * https://github.com/newren/git-filter-repo/issues/87#issue-616956065
 * https://github.com/newren/git-filter-repo/issues/104#issue-630724338
 * https://github.com/newren/git-filter-repo/issues/124#issuecomment-695109714
 * https://github.com/newren/git-filter-repo/issues/189#issue-782155667
 * https://github.com/newren/git-filter-repo/issues/193#issue-791024768

pip should not be used unless you are comfortable debugging _on your own_; do not file requests for help installing with it.  We're extremely worried that pipx will have some of the same problems, but are reluctantly mentioning it as an option since some users have had good luck with it, but if we start getting reports about problems with it we'll rip it out of this install guide too.
```

That's probably too overbearing, but there's no way I'm merging anything claiming that pip works well, since it just results in lots of bug reports from users about how it doesn't.  I'm also worried that mentioning pipx will cause some users who don't have pipx installed but do have pip to think they can just install with pip, but not understand the differences and run into problems, so I really want any mention of pipx to come with a strong disclaimer against using pip.

Even with both of those I'm a bit worried, because marking files as executable was not the only problem folks had with pip, but it's the only one I think pipx will solve.  Granted, that was a big issue, but it's still only one.  That said, I'm willing to give it a try, but not if pipx is highly recommended nor if it intentionally or accidentally comes with an endorsement of using pip to install git-filter-repo.

Reply:
And, of course I was trying to comment on lines 160-166, and the "first line" I referred to was line 160, but GitHub unhelpfully only prints lines 163-166 for context when showing this review on the main page.  Ugh.

Reply:
Python on Windows is slightly different than on Unix. There is no python3 or pip3 and also the shebang line is mainly a unix feature (although the [launcher](https://docs.python.org/3/using/windows.html#python-launcher-for-windows) interprets it). All issues linked above are the result from not knowing the basics of Python on Windows. They are not related to pip or to your project. Exactly he same problems would be observed e.g. when installing poetry. Poetry also suggests [installing via pipx](https://python-poetry.org/docs/).

Reply:
Oh, right, you did mention `into a virtual environment`.  I should read more carefully; sorry about that.  (Though that does suggest it was too easy to miss.)

I agree that the problems were related to users "not knowing the basics of Python on Windows", but the problem was I shouldn't be responsible for educating users on those basics; that should be a thing for the Python project.  But adding instructions to INSTALL.md suggesting users can install via pip (as I did at request of users in the past), resulted in _more_ complaints about the difficult of installing from Windows users and somehow just confused them worse and they didn't go to Python to learn how to do those basics, they asked all their questions of this single-developer (mostly unfunded) project.  At one point I wrote in a commit message:
```
Be more explicit about how pip is lame and provides virtually no benefit
since it leaves you to fix your $PATH yourself, which was the _only_ step
that was needed in installing the whole package anyway.
```
before I later simply removed pip from INSTALL.md entirely (which resulted in getting fewer complaints about the difficulty of installing).

Anyway, thanks for listening to me rant.  While I'm totally unwilling to entertain pip any further (even in a virtual environment), I'm willing to give pipx a try.  I'll look forward to your updated patch.